### PR TITLE
feat(range-calendar): add a new range calendar component

### PR DESCRIPTION
**Pull request by alice-sevin**
Add a new range-calendar component to select a range between two dates display on two calendar one for start one for end
To have same behaviour that range in v-calendar we use v-calendar attributes to highlight dates between start and end in the calendars
Also added an availableDates logic to disable dates that are not able to be selected due to range side (start, end)

Design:
range

Usage:
range

### CODE REVIEW

**Comment by alice-sevin** [comment]
[Composant RangeCalendar](https://trello.com/c/5P2iqZdw/2381-composant-rangecalendar)

File: tests/unit/range-calendar.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      const calendars = wrapper.findAll('Calendar-stub');
+      calendars.at(1).vm.$emit('input', date);
+      expect(wrapper.emitted('input')[0][0]).toStrictEqual({ ...value, end: date });
+    });
+    it('should remove property if Calendar updated value is undefined', () => {
+      const date = undefined;
+      const calendars = wrapper.findAll('Calendar-stub');
+      calendars.at(1).vm.$emit('input', date);
+      expect(wrapper.emitted('input')[0][0]).toStrictEqual({ start: value.start });
+    });
```
Comment (austil) [inline tests/unit/range-calendar.spec.ts:59]: Comment il se produit se cas là ? Le composant calendar permet de désélectionner une valeur ?

**Review by austil — APPROVED** [review]
"Nice and tidy" :+1: 
Ready for merge :green_circle:

File: tests/unit/range-calendar.spec.ts (hunk 1 above)
Comment (alice-sevin) [reply tests/unit/range-calendar.spec.ts:59]: Yes exactement

File: src/lib/dates.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export type DateRange = { start?: Date; end?: Date };
+export type DateRangeSide = 'start' | 'end';
```
[the remark this answers is not in the document]
Comment (alice-sevin) [reply src/lib/dates.ts:2]: Toutafé

File: tests/unit/range-calendar.spec.ts (hunk 1 above)
Comment (alice-sevin) [reply tests/unit/range-calendar.spec.ts:59]: J'ai mis un commentaire pour expliciter côté composant ;)

**Pull request merged**